capsules/extra/src/nonvolatile_storage_driver.rs

```diff
@@ -552,9 +1298,16 @@ impl SyscallDriver for NonvolatileStorage<'_> {
             0 => CommandReturn::success(),
 
             1 => {
-                // How many bytes are accessible from userspace
-                // TODO: Would break on 64-bit platforms
-                CommandReturn::success_u32(self.userspace_length as u32)
+                // How many bytes are accessible to each app
+                let res = self.apps.enter(processid, |app, _kernel_data| app.region);
+
+                // handle case where we fail to enter grant
+                res.map_or(CommandReturn::failure(ErrorCode::NOMEM), |region| {
+                    // handle case where app's region is not assigned
+                    region.map_or(CommandReturn::failure(ErrorCode::FAIL), |region| {
+                        CommandReturn::success_u32(region.length as u32)
+                    })
```

Comment:
> I'm trying to think through this. The region size is fixed by the kernel. Should this just return that size, unconditionally? I guess this is related to the init question.

This should not return the fixed size unconditionally. Each region could potentially have a different size (if it was created with a previous kernel version). So, to maintain backwards compatibility we need to get the region size from what the region header reports.

> Because this is sync this cannot actually init. So we don't actually know if the app will get any space if it has not already been allocated space.

I was considering changing this to be an async function to allow it to complete the `init` operation. However, that's a breaking change to the syscall interface and requires changing existing user apps.

I like the idea of keeping it sync and having it fail immediately, however I feel like this will break some common scenarios.

An app might call this syscall as soon as it starts up to see how much space it has available (to allocate enough space for buffers). The syscall will fail since the app didn't previously call one of the other syscalls to trigger the initialization procedure. If the initialization Then the app will _have_ to know to call one of the other syscalls to kick off the init.